Assets/Scripts/Player/Controller.cs

```diff
@@ -6,45 +6,117 @@ namespace Player
 {
     public class Controller : MonoBehaviour
     {
+        #region Private Variables
+        
         private BaseState _currentState;
+        private InputHandler _inputHandler;
+        
+        #endregion
 
-        public InputHandler InputHandler { get; private set; }
+        #region Interface Variables
+        
         public Rigidbody2D Rigidbody { get; private set; }
         public Animator Animator { get; private set; }
-
+        
         public bool IsGrounded { get; private set; }
-        public bool IsFacingRight { get; set; }
+        public bool IsFacingRight { get; private set; }
```

Comment:
Can you add a TODO comment for this property: When isFacingRight changes (setter) then we also need to either flip the sprite of the active weapon or change the scale of the active weapon gameobject